math/vecops/inc/ROOT/RVec.hxx

```diff
@@ -271,160 +259,1076 @@ hpt->Draw();
+
+   ~RVec()
+   {
+      if (this->Owns) {
+         // Destroy the constructed elements in the vector.
+         this->destroy_range(this->begin(), this->end());
+      }
+   }
+
+   explicit RVec(size_t Size, const T &Value = T()) : SmallVectorImpl<T>(N) { this->assign(Size, Value); }
+
+   template <typename ItTy,
+             typename = typename std::enable_if<std::is_convertible<
+                typename std::iterator_traits<ItTy>::iterator_category, std::input_iterator_tag>::value>::type>
+   RVec(ItTy S, ItTy E) : SmallVectorImpl<T>(N)
+   {
+      this->append(S, E);
+   }
+
+   RVec(std::initializer_list<T> IL) : SmallVectorImpl<T>(N) { this->assign(IL); }
+
+   RVec(const RVec &RHS) : SmallVectorImpl<T>(N)
+   {
+      if (!RHS.empty())
+         SmallVectorImpl<T>::operator=(RHS);
+   }
+
+   RVec &operator=(const RVec &RHS)
    {
-      std::swap(fData, v.fData);
+      SmallVectorImpl<T>::operator=(RHS);
       return *this;
    }
 
-   RVec<T> &operator=(std::initializer_list<T> ilist)
+   RVec(RVec &&RHS) : SmallVectorImpl<T>(N)
    {
-      fData = ilist;
+      if (!RHS.empty())
+         SmallVectorImpl<T>::operator=(::std::move(RHS));
+   }
+
+   RVec(SmallVectorImpl<T> &&RHS) : SmallVectorImpl<T>(N)
+   {
+      if (!RHS.empty())
+         SmallVectorImpl<T>::operator=(::std::move(RHS));
+   }
+
+   RVec(const std::vector<T> &RHS) : RVec(RHS.begin(), RHS.end()) {}
+
+   RVec &operator=(RVec &&RHS)
+   {
+      SmallVectorImpl<T>::operator=(::std::move(RHS));
       return *this;
    }
 
-   // conversion
-   template <typename U, typename = std::enable_if<std::is_convertible<T, U>::value>>
-   operator RVec<U>() const
+   RVec(typename RVec<T>::pointer p, typename RVec<T>::size_type n) : SmallVectorImpl<T>(N)
    {
-      RVec<U> ret(size());
-      std::copy(begin(), end(), ret.begin());
-      return ret;
+      this->BeginX = p;
+      this->fSize = n;
+      this->Capacity = n;
+      this->Owns = false;
    }
 
-   // accessors
-   reference at(size_type pos) { return fData.at(pos); }
-   const_reference at(size_type pos) const { return fData.at(pos); }
-   /// No exception thrown. The user specifies the desired value in case the RVec is shorter than `pos`.
-   value_type at(size_type pos, value_type fallback) { return pos < fData.size() ? fData[pos] : fallback; }
-   /// No exception thrown. The user specifies the desired value in case the RVec is shorter than `pos`.
-   value_type at(size_type pos, value_type fallback) const { return pos < fData.size() ? fData[pos] : fallback; }
-   reference operator[](size_type pos) { return fData[pos]; }
-   const_reference operator[](size_type pos) const { return fData[pos]; }
+   RVec &operator=(SmallVectorImpl<T> &&RHS)
+   {
+      SmallVectorImpl<T>::operator=(::std::move(RHS));
+      return *this;
+   }
+
+   RVec &operator=(std::initializer_list<T> IL)
+   {
+      this->assign(IL);
+      return *this;
+   }
+
+   using SmallVectorTemplateCommon<T>::operator[];
```

Comment:
This using declaration may be the culprit? We have a rather complicated past in terms of using declarations and jitting ;)